src/syscalls/interfaces/uhyve.rs

```diff
@@ -1,53 +1,89 @@
 use memory_addresses::VirtAddr;
+use uhyve_interface::parameters::ExitParams;
+use uhyve_interface::{Hypercall, HypercallAddress};
+#[cfg(target_arch = "x86_64")]
+use x86::io::*;
 
 use crate::arch;
 use crate::arch::mm::paging;
 use crate::syscalls::interfaces::SyscallInterface;
 
-const UHYVE_PORT_EXIT: u16 = 0x540;
+#[inline]
+/// calculates the physical address of the struct passed as reference.
+fn data_addr<T>(data: &T) -> u64 {
```

Comment:
Nit regarding all doc comments: the doc comments should come before attributes. Also, the beginning of doc comments should be capitalized.